src/MICore/PlatformUtilities.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace MICore
+{
+    public static class PlatformUtilities
+    {
+        private enum RuntimePlatform
+        {
+            Unset = 0,
+            Unknown,
+            Windows,
+            MacOSX,
+            Unix,
+        }
+
+        private static RuntimePlatform _runtimePlatform;
+
+        private static RuntimePlatform GetRuntimePlatform()
+        {
+            if (_runtimePlatform == RuntimePlatform.Unset)
+            {
+                _runtimePlatform = CalculateRuntimePlatform();
+            }
+            return _runtimePlatform;
+        }
+
+#if !CORECLR
+        [DllImport("libc")]
+        static extern int uname(IntPtr buf);
+
+        private static RuntimePlatform GetUnixVariant()
+        {
+            IntPtr buf = Marshal.AllocHGlobal(8192);
+            try
+            {
+                if (uname(buf) == 0)
+                {
+                    string os = Marshal.PtrToStringAnsi(buf);
+                    switch (os)
+                    {
+                        case "Darwin":
```

Comment:
Will this will be flagged as an fxcop violation. Should use string comparison here?